### PR TITLE
[EPMRPP-60810] Refactoring of manual deletion of Projects/ Launches/ Item( Suites/Tests/Steps)

### PR DESCRIPTION
**Pull request by pbortnik**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/launch/impl/DeleteLaunchHandlerImpl.java [hunk 1]
```diff
@@ -121,7 +121,7 @@ public DeleteBulkRS deleteLaunches(DeleteBulkRQ deleteBulkRQ, ReportPortalUser.P
 
 		launchRepository.deleteAll(toDelete);
 		toDelete.stream().map(TO_ACTIVITY_RESOURCE).forEach(it -> {
-			eventPublisher.publishEvent(new DeleteLaunchAttachmentsEvent(it.getId()));
+			attachmentRepository.moveForDeletionByLaunchId(it.getId());
 			messageBus.publishActivity(new LaunchDeletedEvent(it, user.getUserId(), user.getUsername()));
```
Comment (miracle8484) [inline src/main/java/com/epam/ta/reportportal/core/launch/impl/DeleteLaunchHandlerImpl.java:125]: Do we really need this sent this event?

File: src/main/java/com/epam/ta/reportportal/core/log/impl/DeleteLogHandlerImpl.java [hunk 2]
```diff
@@ -85,7 +84,8 @@ public OperationCompletionRS deleteLog(Long logId, ReportPortalUser.ProjectDetai
 		Log log = validate(logId, user, projectDetails);
 		try {
 			logRepository.delete(log);
-			cleanUpLogData(log);
+			ofNullable(log.getAttachment()).ifPresent(attachment -> attachmentRepository.moveForDeletion(Collections.singletonList(
+					attachment.getId())));
```
Comment (miracle8484) [inline src/main/java/com/epam/ta/reportportal/core/log/impl/DeleteLogHandlerImpl.java:88]: May be little update and add appropriate method to repository will be better
if (log.getAttachment())  {
    attachmentRepository.moveForDeletion(log.getAttachment().getId());
}

Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/core/log/impl/DeleteLogHandlerImpl.java]: Method replaced with more appropriate

File: src/main/java/com/epam/ta/reportportal/core/launch/impl/DeleteLaunchHandlerImpl.java (hunk 1 above)
Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/core/launch/impl/DeleteLaunchHandlerImpl.java]: Event doesn't relate to deletion. It is logic of storing reportportal activities.
Method is replaced with more appropriate.

Comment (miracle8484) [reply src/main/java/com/epam/ta/reportportal/core/launch/impl/DeleteLaunchHandlerImpl.java]: It depends of the point of view, i thinks if we implement new logic, it should be clean for us - it's deletions from database + moving data to "deletion table". But if we have some extensible point we may have some unpredictable behavior on it. And for now, when we are trying to simplify this flow and improve performance (it's important, we try to simplify, not to make more extensible and so on). Will be better if we remove all unused events(if it's unused) too.

Comment (miracle8484) [reply src/main/java/com/epam/ta/reportportal/core/launch/impl/DeleteLaunchHandlerImpl.java]: Discussed

**Review by miracle8484 — APPROVED** [review]

**Pull request merged**